Storage/ThirdParty/SwiftData.swift

```diff
@@ -137,23 +130,8 @@ open class SwiftData {
      * close a database connection and run a block of code inside it.
      */
     func withConnection<T>(_ flags: SwiftData.Flags, synchronous: Bool = false, _ callback: @escaping (_ connection: SQLiteDBConnection) throws -> T) -> Deferred<Maybe<T>> {
```

Comment:
The synchronous parameter is no longer used here but I want to keep the change as simple as possible in case there is a need to revert. If this lands successfully I'll come back later to clean up the interfaces.

Reply:
So the whole idea behind secondaryConnection is to support more than one db queue? 
What happens with multiple queues 🤔

Reply:
My understand (which may be incorrect) is that it was to allow concurrent db queries to be run. Removing it greatly simplifies reasoning about what's happening at a given time but will potentially result in slower db queries due to them now being run serially.

Reply:
This is indeed controversial change as on one side you simply things and it has the potential to resolve the bug with concurrency issue related to the db. On the other hand having removed this I am in the dark on what impact it will have. 

We average around 92~100MB of db write per day. Which isn't that high and the highest is around 500MB for iPhone(s). 

Could we find areas we might experience an impact and test around it if it even solves the bug and that there is no db impact?
- Bookmars
- Favicons
- History 
- Top sites
... etc

Reply:
I believe reading list and top sites are the only use cases that will be impacted by this, as everything else has been moved to be managed by the A/S layer.

Reply:
I would ship in nightly and ask QA to keep an eye out for any hangs / slowness in the app around that area

Reply:
@lougeniaC64 any objections here or something you might think that this might break?

@OrlaM consider it an R+ from my end, will wait for @lougeniaC64's assessment as well

Reply:
@nbhasin2 This will be released after the rust sync manager integration so this shouldn't directly impact the storage for our components as they don't have a BrowserDB dependency. However, I think there are a few tables in BrowserDB that are adjacent to or reliant on our components (Top Sites for instance). cc @tarikeshaq in the even this isn't true for places or the upcoming push integration.

But in short, I'm okay with Orla's approach of pushing these changes to Beta and observing the impact.

Reply:
+1 from me. For Top Sites, I believe it's only pinned sites that get persisted there so that should be a lower risk of performance problems unless the connection gets a lot of contention, plus landing and observing sounds like the best way without speculation